src/main/java/edu/kis/powp/jobs2d/command/CommandBoundariesCheckVisitor.java

```diff
@@ -4,11 +4,11 @@
 
 import static java.lang.Math.abs;
 
-public class CommandBoundariesCheckVisitor implements Visitor{
-	private final ICanvas canvas;
+public class CommandBoundariesCheckVisitor implements Visitor, ICommandBoundariesCheckStrategy{
```

Comment:
Implementing ICommandBoundariesCheckStrategy is closer to the Template Method pattern approach. In Strategy pattern ICommandBoundariesCheckStrategy would be an attribute. Please refactor.